### PR TITLE
DRILL-8149: large xlsx configs

### PR DESCRIPTION
**Pull request by pjfanning**
## Description

https://issues.apache.org/jira/browse/DRILL-8149

## Documentation
(Please describe user-visible changes similar to what should appear in the Drill documentation.)

## Testing
(Please describe how this PR has been tested.)

### CODE REVIEW

**Comment by pjfanning** [comment]
Still needs test coverage

File: contrib/format-excel/src/main/java/org/apache/drill/exec/store/excel/ExcelBatchReader.java [hunk 1]
```diff
@@ -230,6 +239,18 @@ private void openFile(FileScanFramework.FileSchemaNegotiator negotiator) {
     try {
       fsStream = negotiator.fileSystem().openPossiblyCompressedStream(split.getPath());
 
+      if (readerConfig.maxArraySize >= 0) {
```
Comment (luocooong) [inline contrib/format-excel/src/main/java/org/apache/drill/exec/store/excel/ExcelBatchReader.java:242]: Is it possible to add the unit tests for these three options? we can refer to the : `TestExcelFormat#testLastRowQuery()`.

File: contrib/format-excel/src/main/java/org/apache/drill/exec/store/excel/ExcelFormatConfig.java [hunk 2]
```diff
@@ -98,6 +109,34 @@ public String getSheetName() {
     return sheetName;
   }
 
+  /**
+   * See the <code>setByteArrayMaxOverride</code> section in the Apache POI
```
Comment (luocooong) [inline contrib/format-excel/src/main/java/org/apache/drill/exec/store/excel/ExcelFormatConfig.java:113]: Is it possible to update the `drill/contrib/format-excel/README.md` file?

**Review by luocooong — COMMENTED** [review]
@pjfanning Thanks for the update. In addition to comments, I also recommend that you add a description for this page because the description in JIRA is valuable, but the reviewer cannot read it directly here.

File: contrib/format-excel/src/main/java/org/apache/drill/exec/store/excel/ExcelBatchReader.java [hunk 3]
```diff
@@ -183,6 +189,9 @@ public String getFieldName() {
       lastColumn = plugin.getConfig().getLastColumn();
       allTextMode = plugin.getConfig().getAllTextMode();
       sheetName = plugin.getConfig().getSheetName();
+      maxArraySize = plugin.getConfig().getMaxArraySize();
```
Comment (cgivre) [inline contrib/format-excel/src/main/java/org/apache/drill/exec/store/excel/ExcelBatchReader.java:192]: @pjfanning Thanks for the update!   Can you update the `README` file in this folder to explain what these options do?  Do you have recommended values?

**Comment by pjfanning** [comment]
@cgivre @luocooong thanks for the feedback. I've been busy on other stuff and am not prioritising completing this. Before it is ready to merge, it will need doc updates and test coverage.

**Comment by cgivre** [comment]
> 

No worries!  It looks good to me other than adding some doc updates.

**Comment by luocooong** [comment]
@pjfanning Is it possible to resolve the above conflicts and do a rebase on the master branch? Because I'm ready to merge this good feature.

**Comment by pjfanning** [comment]
> @pjfanning Is it possible to resolve the above conflicts and do a rebase on the master branch? Because I'm ready to merge this good feature.

I've resolved the merge conflict but am leaving this in draft because I still need to add test coverage.

**Comment by pjfanning** [comment]
@luocooong @cgivre I'm not sure how unit-testable these changes are. To really test the features would require very large xlsx files and even with the all the new settings in place, the tests would be slow to run and use a lot of memory. Would it be feasible to treat these properties as something that users would only need to set in rare circumstances? If users run into issues with the properties in real world scenarios, then maybe the code and tests can be beefed up then.

**Comment by cgivre** [comment]
> @luocooong @cgivre I'm not sure how unit-testable these changes are. To really test the features would require very large xlsx files and even with the all the new settings in place, the tests would be slow to run and use a lot of memory. Would it be feasible to treat these properties as something that users would only need to set in rare circumstances? If users run into issues with the properties in real world scenarios, then maybe the code and tests can be beefed up then.

I'm ok with not unit testing these.  Since all that is happening here is that that we are creating some new config variables in the format config and passing them down to a reader, I think it is fine.  So, unless there is any objection, LGTM +1.

**Pull request merged**